core/src/main/java/org/infinispan/commands/write/InvalidateL1Command.java

```diff
@@ -104,8 +104,8 @@ public boolean shouldInvoke(InvocationContext ctx) {
       for (Object k : getKeys()) {
          // If any key in the set of keys to invalidate is not local, or we are uncertain due to a rehash, then we
          // process this command.
-         DataLocality locality = dm.getLocality(k);
-         if (!locality.isLocal() || locality.isUncertain()) return true;
+         if (!dm.getDistributionInfo(k).read().isOwner())
```

Comment:
True, going through all the interceptors isn't that much extra work, and removing the check here would allow us to remove `VisitableCommand.shouldInvoke()` as well. I've created [ISPN-7431](https://issues.jboss.org/browse/ISPN-7431) for it.